devices/dragonboard/dragonboard.py

```diff
@@ -259,31 +257,34 @@ def flash_test_image(self, server_ip, server_port):
         try:
             # XXX: I hope 30 min is enough? but maybe not!
             self._run_control(cmd, timeout=1800)
-        except:
+        except subprocess.TimeoutExpired:
             raise ProvisioningError("timeout reached while flashing image!")
         try:
             self._run_control('sync')
-        except:
+        except subprocess.SubprocessError:
             # Nothing should go wrong here, but let's sleep if it does
             logger.warn("Something went wrong with the sync, sleeping...")
             time.sleep(30)
         try:
             self._run_control(
                 'sudo hdparm -z {}'.format(self.config['test_device']),
                 timeout=30)
-        except:
-            raise ProvisioningError("Unable to run hdparm to rescan "
-                                    "partitions")
+        except subprocess.CalledProcessError as exc:
+            raise ProvisioningError(
+                "Unable to run hdparm to rescan"
+                "partitions: {}".format(exc.output))
```

Comment:
is this (and the other places with exc.output) a place where chaining exceptions would be more natural than getting the output from the caught exception?

Reply:
I'm not really sure. First - this was a bit of an opportunistic fix. I didn't previously capture the output at all, but when I was looking at it, I thought, "hey, I could easily catch the output there too!" So in general, I'm trying to generalize the exception as a ProvisioningError for $reasons but capturing the output is just bonus